fvtest/coretest/TestTypeTraits.cpp

```diff
@@ -0,0 +1,121 @@
+#include <OMR/TypeTraits.hpp>
+#include <gtest/gtest.h>
+
+namespace OMR
+{
+
+TEST(TestTypeTraits, IntegralConst)
+{
+	EXPECT_EQ(0, (IntegralConstant<int, 0>::VALUE));
+	EXPECT_EQ(0xdeadbeef, (IntegralConstant<unsigned int, 0xdeadbeef>::VALUE));
+}
+
+TEST(TestTypeTraits, BooleanConstant)
+{
+	EXPECT_TRUE((BoolConstant<true>::VALUE));
+	EXPECT_TRUE(TrueConstant::VALUE);
+
+	EXPECT_FALSE((BoolConstant<false>::VALUE));
+	EXPECT_FALSE(FalseConstant::VALUE);
+}
+
+TEST(TestTypeTraits, IsSame)
+{
+	EXPECT_TRUE((IsSame<int, int>::VALUE));
+	EXPECT_FALSE((IsSame<int, double>::VALUE));
+	EXPECT_TRUE((IsSame<void*, void*>::VALUE));
+	EXPECT_FALSE((IsSame<void* const, void*>::VALUE));
+	EXPECT_FALSE((IsSame<void*, const void*>::VALUE));
+}
+
+TEST(TestTypeTraits, TypeAlias)
+{
+	EXPECT_TRUE((IsSame<void, TypeAlias<void>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<void*, TypeAlias<void*>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<const int, TypeAlias<const int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int&, TypeAlias<int&>::Type>::VALUE));
+}
+
+TEST(TestTypeTraits, RemoveConst)
+{
+	EXPECT_TRUE((IsSame<int, RemoveConst<int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int, RemoveConst<const int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<volatile int, RemoveConst<const volatile int>::Type>::VALUE));
+}
+
+TEST(TestTypeTraits, RemoveVolatile)
+{
+	EXPECT_TRUE((IsSame<int, RemoveVolatile<int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int, RemoveVolatile<volatile int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<const int, RemoveVolatile<const volatile int>::Type>::VALUE));
+}
+
+TEST(TestTypeTraits, RemoveCv)
+{
+	EXPECT_TRUE((IsSame<int, RemoveCv<int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int, RemoveCv<const int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int, RemoveCv<const volatile int>::Type>::VALUE));
+}
+
+TEST(TestTypeTraits, RemoveReference)
+{
+	EXPECT_TRUE((IsSame<int, RemoveReference<int>::Type>::VALUE));
+	EXPECT_TRUE((IsSame<int, RemoveReference<int&>::Type>::VALUE));
```

Comment:
It might be worth to also add:

```c++
	EXPECT_TRUE((IsSame<const int, RemoveReference<const int&>::Type>::VALUE)); // or volatile
	EXPECT_TRUE((IsSame<int*, RemoveReference<int*&>::Type>::VALUE));
```

Reply:
added.